ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -66,48 +67,140 @@ func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error)
+// GetConfigs gets the ecs-cli config object from the config file(s).
 // map contains the keys that are present in the config file (maps string field name to string field value)
 // map is type map[interface{}]interface{} to ensure fowards compatibility with changes that will
 // cause certain keys to be mapped to maps of keys
-func (rdwr *YAMLReadWriter) GetConfig() (*CLIConfig, map[interface{}]interface{}, error) {
-	to := new(CLIConfig)
+// This function either reads the old single configuration file
+// Or if the new files are present, it reads from them instead
+func (rdwr *YAMLReadWriter) GetConfigs(clusterConfig string, profileConfig string) (*CLIConfig, map[interface{}]interface{}, error) {
+	cliConfig := &CLIConfig{}
 	configMap := make(map[interface{}]interface{})
 	// read the raw bytes of the config file
-	path := configPath(rdwr.destination)
+	profilePath := credentialsFilePath(rdwr.destination)
+	configPath := configFilePath(rdwr.destination)
+
+	// Try to read the config as YAML
+	clusterMap := make(map[interface{}]interface{})
+	errYAMLConfig := readYAML(configPath, clusterMap)
+	if errYAMLConfig == nil {
+		// cluster config exists and is YAML
+		processClusterMap(clusterConfig, clusterMap, configMap, cliConfig)
+	}
 
-	if errYAML := readYAML(path, configMap, to); errYAML == nil {
-		// File is YAML
-		return to, configMap, nil
-	} else if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
+	if _, err := os.Stat(profilePath); err != nil {
+		// credentials file exists- so that means we are using the new style configs
+		profileMap := make(map[interface{}]interface{})
+		if err = readYAML(profilePath, profileMap); err != nil {
+			return nil, nil, errors.Wrapf(err, "Error reading or parsing credentials file, %s", profilePath)
+		}
+
+		if err = processProfileMap(profileConfig, profileMap, configMap, cliConfig); err != nil {
+			return nil, nil, errors.Wrapf(err, "Error processing credentials file, %s", profilePath)
+		}
+		return cliConfig, configMap, nil
+	}
+
+	if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
 		// File is INI
 		return to, configMap, nil
-	} else if errINI != nil {
-		// Return the ini error
-		// This will return a parsing error for ini (format error)
-		return nil, nil, errINI
-	} else {
-		// If yaml through an error, but ini didn't throw an error- return the yaml error
-		return nil, nil, errYAML
 	}
+
+	if _, err := os.Stat(configPath); err != nil {
+		// Cluster config file exists, but it wasn't INI, so that means it was YAML but there was a parse error
+		return nil, nil, errYAMLConfig
+	}
+
+	// if no configs exist, we return empty objects
+	return cliConfig, configMap, nil
+
+}
+
+func processProfileMap(profileKey string, profileMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if profileKey == "" {
+		var ok bool
+		if profileKey, ok = profileMap["default"].(string); !ok {
```

Comment:
So profileKey is the name of the profile that we want to read. Either is actually the name of a key in the config, or it is empty, which means that we want to find the defined default and use that as the key. If you guys think it would make the code better style-wise- it could work like this:

```
var string profile
if profileKey == ""
    profile = default
else
    profile = profileKey

But my way uses less lines of code! :P